java/ql/src/utils/model-generator/SinkFlows.ql

```diff
@@ -0,0 +1,40 @@
+/**
+ * @name Capture sink models.
+ * @description Finds public methods that act as sinks as they flow into a known sink.
+ * @kind path-problem
+ * @id java/utils/model-generator/sink-models-path-problem
+ * @severity info
+ * @tags model-generator
+ */
+
+import utils.modelgenerator.internal.CaptureModels as CaptureModels
+import utils.modelgenerator.internal.CaptureModelsSpecific as CaptureModelsSpecific
+import java
+import DataFlow::PathGraph
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.ExternalFlow as ExternalFlow
+
+class Activate extends CaptureModels::ActiveConfiguration {
+  override predicate activateToSinkConfig() { any() }
+}
+
+class PropagateToSinkConfiguration extends CaptureModels::PropagateToSinkConfiguration {
+  PropagateToSinkConfiguration() { this = "parameters or fields flowing into sinks" }
+
+  override predicate sinkGrouping(DataFlow::Node sink, string sinkGroup) {
+    ExternalFlow::sinkNode(sink, sinkGroup)
+  }
+}
+
+from
```

Comment:
Is this query for debugging purposes?

Reply:
Yes it is. Do you think we should move it somewhere else? I have no clue what side effects it might have if we have it in `java/ql/src`.

Reply:
I guess I could move it to `ql/java/ql/test/utils/model-generator/`

Reply:
It should be fine to keep it here, we just need to make sure that it is excluded from the query suites, which should be the case when the `model-generator` tag is used.
In any case, maybe we should consider 
- Introducing a new tag named `debug` and also tag the query with that to signal that it is *only* used for debugging purposes (this requires that we add it to the exclusion list in `security-and-quality-selectors.yml` AND the other filters).
- Make a debug folder in the model-generator folder and put the queries there.

Also, we might get an issue with these queries (I have a draft PR somewhere doing approximately the same as you are doing here, but this might not be compatible with the field summary generation).

Reply:
I pushed some changes with the `debug` tag. (I also squashed and rebased the PR to fix merge conflicts) Let me know if something is missing.

Reply:
That part looks good to me!